### PR TITLE
Additional google analytics events

### PR DESCRIPTION
**Pull request by NoahCarnahan**
Add google analytics events for the following:
- Change case list/detail property format
- Download Form Export
- Download Case Export

(The commit list here is confusing, sorry bout that)

### CODE REVIEW

File: corehq/apps/reports/static/reports/ko/export.manager.js [hunk 1]
```diff
@@ -35,12 +35,24 @@ var ExportManager = function (o) {
...
+         *  - xmlns
+         * the two optional keys are used for google analytics.
+         * @param params
+         */
         updateModal = function(params) {
             var autoRefresh = '';
             var pollDownloader = function () {
-                if ($('#ready_'+params.response.download_id).length == 0) {
-                    $.get(params.response.download_url, function(data) {
+                if ($('#ready_'+params.data.download_id).length == 0) {
```
Comment (dimagimon) [inline corehq/apps/reports/static/reports/ko/export.manager.js]: Use '===' to compare with '0'.

**Comment by NoahCarnahan** [comment]
Waiting for a confirmation on the spec

File: corehq/apps/reports/static/reports/javascripts/reports.download_export.js [hunk 2]
```diff
@@ -36,4 +37,14 @@ var HQExportDownloader = function (options) {
             });
         });
     };
+    /**
+     * Attach a click event handler to the download modal "download" button that
+     * notifies google of the download event.
+     */
+    self.setUpEventTracking = function(){
```
Comment (NoahCarnahan) [inline corehq/apps/reports/static/reports/javascripts/reports.download_export.js]: You might notice that this function is very similar to the identically named `setUpEventTracking` function on `ExportManager` object. This object (`HQExportDownloader`) duplicates some functionality of `ExportManager` and exists on the same page. `HQExportDownloader` should probably be done away with. Seems beyond the scope of this feature though.

File: corehq/apps/reports/static/reports/ko/export.manager.js [hunk 3]
```diff
@@ -54,14 +66,6 @@ var ExportManager = function (o) {
...
                 clearInterval(autoRefresh);
             });
-            if (params.xmlns == "http://code.javarosa.org/devicereport"){
-                gaTrackLink(
-                    $(".download-button", self.exportModal),
-                    "Form Exports",
-                    "Download Mobile Device Log",
-                    "Export Mobile Device Log"
-                );
-            }
```
Comment (NoahCarnahan) [inline corehq/apps/reports/static/reports/ko/export.manager.js]: This block has been moved into the `setUpEventTracking` function.

File: corehq/apps/reports/templates/reports/reportdata/case_export_data.html [hunk 4]
```diff
@@ -79,7 +80,11 @@
                         </a>
                     </td>
                     <td class="cell-vertical-centered">
-                        <a data-toggle="modal" href="#export-download-status" data-backdrop="static" data-formname="{{ export.name }}" data-dlocation='{% url "export_custom_data" domain export.get_id %}?type=case&{{ get_filter_params.urlencode|safe }}&async=true'
+                        <a data-toggle="modal"
+                           href="#export-download-status"
+                           data-backdrop="static"
+                           data-formname="{{ export.name }}"
+                           data-dlocation='{% url "export_custom_data" domain export.get_id %}?type=case&{{ get_filter_params.urlencode|safe }}&async=true'
```
Comment (NoahCarnahan) [inline corehq/apps/reports/templates/reports/reportdata/case_export_data.html]: (just making this one more readable)

**Comment by nickpell** [comment]
looks good to merge when spec is confirmed.

File: corehq/apps/reports/static/reports/ko/export.manager.js [hunk 5]
```diff
@@ -35,12 +36,24 @@ var ExportManager = function (o) {
...
+         *  - xmlns
+         * the two optional keys are used for google analytics.
+         * @param params
+         */
         updateModal = function(params) {
             var autoRefresh = '';
             var pollDownloader = function () {
-                if ($('#ready_'+params.response.download_id).length == 0) {
-                    $.get(params.response.download_url, function(data) {
+                if ($('#ready_'+params.data.download_id).length == 0) {
```
Comment (dimagimon) [inline corehq/apps/reports/static/reports/ko/export.manager.js]: Use '===' to compare with '0'.

File: corehq/apps/reports/static/reports/ko/export.manager.js [hunk 6]
```diff
@@ -83,6 +88,50 @@ var ExportManager = function (o) {
...
+            // Device reports event
+            // (This is a bit of a special case due to its unique "action" and "label"
+            if (params.xmlns == "http://code.javarosa.org/devicereport") {
+                gaTrackLink(downloadButton, "Form Exports", "Download Mobile Device Log", "Export Mobile Device Log");
+            }
+
+            var category, action;
+            if (self.export_type == "form"){
+                category = "Form Exports";
+                action = "Download (any) Form Export"
```
Comment (dimagimon) [inline corehq/apps/reports/static/reports/ko/export.manager.js]: Missing semicolon.

File: corehq/apps/reports/static/reports/ko/export.manager.js (hunk 5 above)
Comment (dimagimon) [inline corehq/apps/reports/static/reports/ko/export.manager.js]: Use '===' to compare with '0'.

File: corehq/apps/reports/static/reports/ko/export.manager.js [hunk 7]
```diff
@@ -35,12 +36,25 @@ var ExportManager = function (o) {
...
+         *  - xmlns
+         * the two optional keys are used for google analytics.
+         * @param params
+         */
         updateModal = function(params) {
             var autoRefresh = '';
             var pollDownloader = function () {
-                if ($('#ready_'+params.response.download_id).length == 0) {
-                    $.get(params.response.download_url, function(data) {
+                if ($('#ready_'+params.data.download_id).length == 0) {
```
Comment (dimagimon) [inline corehq/apps/reports/static/reports/ko/export.manager.js]: Use '===' to compare with '0'.

File: corehq/apps/reports/static/reports/ko/export.manager.js [hunk 8]
```diff
@@ -83,6 +89,44 @@ var ExportManager = function (o) {
...
+                    var label = $('#include-closed-select').val() == "true" ? "all" : "all open";
+                    gaTrackLink(downloadButton, "Download Case Export", "Download Raw Case Export", label);
+                }
+
+            } else if (self.export_type == "form"){
+                var category = "Download Form Export";
+
+
+                var action = "Download Raw Form Export";
+                var label = params.isBulkDownload ? "bulk" : params.formName;
```
Comment (dimagimon) [inline corehq/apps/reports/static/reports/ko/export.manager.js]: 'label' is already defined.

File: corehq/apps/reports/static/reports/ko/export.manager.js [hunk 9]
```diff
@@ -35,12 +36,26 @@ var ExportManager = function (o) {
...
+         *  - exportName
+         * the two optional keys are used for google analytics.
+         * @param params
+         */
         updateModal = function(params) {
             var autoRefresh = '';
             var pollDownloader = function () {
-                if ($('#ready_'+params.response.download_id).length == 0) {
-                    $.get(params.response.download_url, function(data) {
+                if ($('#ready_'+params.data.download_id).length == 0) {
```
Comment (dimagimon) [inline corehq/apps/reports/static/reports/ko/export.manager.js]: Use '===' to compare with '0'.

**Comment by NoahCarnahan** [comment]
@nickpell spec has finally settled :) This should be good to go.

**Pull request merged**